Re-adding Scripts for Code Review

DavidJB526 opened the pull request.

DavidJB526 committed:
Re-adding Scripts for Code Review

dantogno commented on Assets/Scripts/BulletController.cs:
```diff
@@ -0,0 +1,15 @@
+using System.Collections.Generic;
+using UnityEngine;
+
+public class BulletController : MonoBehaviour {
```
Put open curly brackets on new lines

DavidJB526 replied:
Done

dantogno commented on Assets/Scripts/BulletController.cs:
```diff
@@ -0,0 +1,15 @@
+
+    private void OnTriggerEnter2D(Collider2D collision)
+    {
+        if (collision.tag != "Player" && collision.tag != "Background" && collision.tag != "Collectable" && collision.tag != "Checkpoint")
```
What's the purpose of this code? It's unclear, so I'd wrap it in a function with a name so I understand the purpose.

dantogno replied:
For long, complex conditions like this, combine them into some local bool variables to help express them clearly and without extending so far to the right. For example `bool isNotPlayerOrBackground = collison.tag !=Player" && collision.tag != "background";`

dantogno commented on Assets/Scripts/CameraController.cs:
```diff
@@ -0,0 +1,23 @@
+using System.Collections.Generic;
+using UnityEngine;
+
+public class CameraController : MonoBehaviour {
```
curly bracket on new line

DavidJB526 replied:
Deleted, using Cinemachine

dantogno commented on Assets/Scripts/CameraController.cs:
```diff
@@ -0,0 +1,23 @@
+
+	// Use this for initialization
+	void Start ()
+    {
```
fix indent

DavidJB526 replied:
Deleted, using Cinemachine

dantogno commented on Assets/Scripts/CameraController.cs:
```diff
@@ -0,0 +1,23 @@
+using System.Collections.Generic;
+using UnityEngine;
+
+public class CameraController : MonoBehaviour {
```
I'd probably use cinemachine and not this script. And then if you're not using this script, delete it.

DavidJB526 replied:
Deleted, using Cinemachine

dantogno commented on Assets/Scripts/Checkpoint.cs:
```diff
@@ -0,0 +1,36 @@
+    private SpriteRenderer spriteRenderer;
+
+	void Start ()
+    {
```
fix indent

DavidJB526 replied:
Done

dantogno commented on Assets/Scripts/Checkpoint.cs:
```diff
@@ -0,0 +1,36 @@
+using System.Collections.Generic;
+using UnityEngine;
+
+public class Checkpoint : MonoBehaviour {
```
curly bracket on new line

DavidJB526 replied:
Done

dantogno commented on Assets/Scripts/Collectable.cs:
```diff
@@ -0,0 +1,22 @@
+    private float rotationSpeed;
+
+	private void FixedUpdate ()
+    {
```
fix indent

dantogno commented on Assets/Scripts/Collectable.cs:
```diff
@@ -0,0 +1,22 @@
+    private void OnTriggerEnter2D(Collider2D collision)
+    {
+        if (collision.CompareTag("Player"))
+        {
```
Add sound effect to collectible, make sure you aren't destroying the collectible too early. Watch and follow my tutorial: https://youtu.be/GTsilz2v_PM

dantogno commented on Assets/Scripts/Door.cs:
```diff
@@ -0,0 +1,20 @@
+        {
+            SceneManager.LoadScene(nextScene);
+        }
+
```
Remove excess whitespace

DavidJB526 replied:
Done

dantogno commented on Assets/Scripts/MouseTracker.cs:
```diff
@@ -0,0 +1,24 @@
+using System.Collections.Generic;
+using UnityEngine;
+
+public class MouseTracker : MonoBehaviour {
```
curly bracket on next line

dantogno replied:
What's this class for? Add a summary (type three `/`'s in visual studio for the shortcut. https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/xmldoc/summary

dantogno commented on Assets/Scripts/MouseTracker.cs:
```diff
@@ -0,0 +1,24 @@
+        angle = Mathf.Atan2(mousePosition.y, mousePosition.x) * Mathf.Rad2Deg;
+        transform.rotation = Quaternion.Euler(new Vector3(0, 0, angle));
+    }
+
```
remove extra whitespace

dantogno commented on Assets/Scripts/PlayerController.cs:
```diff
@@ -0,0 +1,227 @@
+public class PlayerController : MonoBehaviour {
+
+    [SerializeField]
+    private float playerAcceleration, maxSpeed, jumpForceForward, jumpForceUp, bulletSpeed, recoilSpeed, overheatAdd, overheatSubtract;
```
Add some ToolTip attributes for these to clarify them. https://docs.unity3d.com/ScriptReference/TooltipAttribute.html

dantogno commented on Assets/Scripts/PlayerController.cs:
```diff
@@ -0,0 +1,227 @@
+    private Text collectableText;
+    [SerializeField]
+    private Slider overheatSlider;
+
```
remove extra whitespace

dantogno commented on Assets/Scripts/PlayerController.cs:
```diff
@@ -0,0 +1,227 @@
+        isDead = false;
+    }
+
+    private void UpdateCollectables()
```
I don't like this name, I'd name it more like "UpdateCollectableText" since you're updating the text, not the collectibles themselves.

dantogno commented on Assets/Scripts/PlayerController.cs:
```diff
@@ -0,0 +1,227 @@
+        }
+        else if (other.CompareTag("Collectable"))
+        {
+            collectables++;
```
I'd move this logic to the collectable class

dantogno commented on Assets/Scripts/PlayerController.cs:
```diff
@@ -0,0 +1,227 @@
+
+    private void UpdateCollectables()
+    {
+        collectableText.text = "Collectables: " + collectables;
```
I'd move this logic to the collectable class, or make a new class just for handling the text. The idea is, if it's something about the collectable class, handle it in the collectable class. Let me know if you have questions :) https://en.wikipedia.org/wiki/Separation_of_concerns

dantogno reviewed (changes requested):
Overall, really nicely written code, great job following the guidelines! 💯 I nitpicked some stuff, but that's what review is for :) Please make the changes indicated, or reply to the particular comments you have questions about. Thanks!

DavidJB526 committed:
Move Bracket to next line

DavidJB526 committed:
Remove CameraController script

Using Cinemachine instead

DavidJB526 committed:
Move Curly Bracket to new Line and Fix Start Curly Bracket Indent

DavidJB526 committed:
Remove Excess Whitespace

DavidJB526 committed:
Fix Indent

Commit:
Move Curly Bracket in MouseTracker to Next Line

Commit:
Remove Extra Whitespace in MouseTracker

Commit:
Add Description to MouseTracker

Commit:
Remove Extra Whitespace in PlayerController

Commit:
Specified Variables in BulletController OnTriggerEnter2D

Commit:
Move UpdateCollectableText to new Class Collectable Manager

Also moved the logic for updating the static variable "collectables" into the Collectable class

Commit:
Add Tooltips to PlayerController Float Variables

Commit:
Add Pickup Sound to Collectable

Commit:
Fix CollectableManager

Commit:
Speed Up Player Animations

Commit:
Fix PhysicsMaterial and Death/Respawn Animator

Commit:
Add Sound to Bullet

Commit:
Add Sound to Checkpoint

Commit:
Add Death and Jump Sounds to Player

dantogno commented on Assets/Animation/Player/Player.controller:
```diff
@@ -13,37 +13,37 @@ AnimatorController:
     m_DefaultInt: 0
     m_DefaultBool: 0
-    m_Controller: {fileID: 0}
+    m_Controller: {fileID: 9100000}
```
What are all these meta files changing?

DavidJB526 replied:
This could be moving between 1.9f1 and 1.9f2, but it shouldn't cause any issue

dantogno commented on Assets/Prefabs/Bullet.prefab:
```diff
@@ -24,6 +24,7 @@ GameObject:
   - component: {fileID: 114438365460165132}
   - component: {fileID: 50789667184417300}
   - component: {fileID: 95691846921612874}
+  - component: {fileID: 82834473738846460}
```
Why is bullet prefab changing?

DavidJB526 replied:
I added an AudioSource to it

dantogno commented on Assets/Scripts/MouseTracker.cs:
```diff
@@ -2,8 +2,12 @@
+/// </summary>
+public class MouseTracker : MonoBehaviour
+{
     private Vector3 mousePosition;
```
Put your serialized fields above your plain old private fields.

dantogno commented on Assets/Scripts/MouseTracker.cs:
```diff
@@ -20,5 +24,4 @@ private void FixedUpdate()
         angle = Mathf.Atan2(mousePosition.y, mousePosition.x) * Mathf.Rad2Deg;
```
This math is black magic as far as I know but I'm dumb 😜.  One thing that's expressed a little unclearly is this mousePosition variable. It originally represents the cursor position as I would expect. But soon you overwrite it's value with the distance between the mosue position and the fire point. I'm not sure why (comment). And once you do that, the new value shouldn't really be called mouse position, because it represents something different now.

dantogno commented on Assets/Scripts/PlayerController.cs:
```diff
@@ -7,7 +7,27 @@
+    [Tooltip("How much Force is applied to the Player when they Shoot")]
+    private float recoilSpeed;
+    [SerializeField]
+    [Tooltip("How much Overheat is Added on each Shot")]
```
Why is Added capitalized?

dantogno commented on Assets/Scripts/PlayerController.cs:
```diff
@@ -7,7 +7,27 @@
+    [Tooltip("How far Forward the Player Jumps on Jump")]
+    private float jumpForceForward;
+    [SerializeField]
+    [Tooltip("How far Up the Player Jumps on Jump")]
```
why is "Jumps" capitalized?

dantogno commented on Assets/Scripts/PlayerController.cs:
```diff
@@ -7,7 +7,27 @@
+    [Tooltip("Max Speed that the Player can Move")]
+    private float maxSpeed;
+    [SerializeField]
+    [Tooltip("How far Forward the Player Jumps on Jump")]
```
If it's a force your'e added, it's not measured in distance, but your tooltip suggests it's a distance. More accurate description would be "how much forward force to apply  when the player jumps". or something of that nature.

dantogno commented on Assets/Scripts/PlayerController.cs:
```diff
@@ -7,7 +7,27 @@
+    [Tooltip("How Fast the Bullets Move")]
+    private float bulletSpeed;
+    [SerializeField]
+    [Tooltip("How much Force is applied to the Player when they Shoot")]
```
So should this number be a negative? At the minimum describe if this should be pos or neg, or you could try using the range attribute to enforce it. https://docs.unity3d.com/ScriptReference/RangeAttribute.html

dantogno commented on Assets/Scripts/PlayerController.cs:
```diff
@@ -7,7 +7,27 @@
+    private float overheatAdd;
+    [SerializeField]
+    [Tooltip("How much Overheat is Subtracted each Physics Frame")]
+    private float overheatSubtract;
```
Same thing, should this be pos or negative?

dantogno commented on Assets/Scripts/PlayerController.cs:
```diff
@@ -7,7 +7,27 @@
+    private float overheatSubtract;
+
     [SerializeField]
     private PhysicsMaterial2D playerMoving, playerStopping;
```
Give this some tooltip love

dantogno commented on Assets/Scripts/PlayerController.cs:
```diff
@@ -21,28 +41,31 @@ public class PlayerController : MonoBehaviour {
-
     private Rigidbody2D rb2d;
     private Animator anim;
+    private AudioSource[] audioSources;
```
The player more than likely only represents a single audio source. Audio sources should be considered like speakers in the scene. You wouldn't need a new set of speakers for every song you want to play at home. You only need new speakers in the positions in the world that are important for positional audio. Since all these sounds in theory are coming from the player, a single audio source component makes sense.

Instead of multiple sources, us multiple clips and a single source. See my next comment below.

dantogno commented on Assets/Scripts/PlayerController.cs:
```diff
@@ -21,28 +41,31 @@ public class PlayerController : MonoBehaviour {
         anim = GetComponent<Animator>();
+        audioSources = GetComponents<AudioSource>();
+
+        jumpSound = audioSources[0];
```
I would make these of the type AudioClip and make them serialized fields and assign them in the editor.

dantogno commented on Assets/Scripts/BulletController.cs:
```diff
@@ -2,13 +2,29 @@
+            spriteRenderer.enabled = false;
+            Destroy(gameObject, audioSource.clip.length);
         }
         
```
remove extra whitespace

dantogno commented on Assets/Scripts/BulletController.cs:
```diff
@@ -2,13 +2,29 @@
+        spriteRenderer = GetComponent<SpriteRenderer>();
+    }
 
     private void OnTriggerEnter2D(Collider2D collision)
```
I'd go ahead and add a summary tag here to explain the intent of the bullet and when it's removed.

dantogno commented on Assets/Scripts/Checkpoint.cs:
```diff
@@ -2,27 +2,32 @@
     private Sprite inactiveSprite, activeSprite;
 
     private bool isActivated;
+
```
remove whitespace

dantogno commented on Assets/Scripts/Collectable.cs:
```diff
@@ -7,16 +7,33 @@ public class Collectable : MonoBehaviour {
     private float rotationSpeed;
 
-	private void FixedUpdate ()
+    public static float collectables = 0f;
```
One thing I forgot to talk about in my video is that static values hang around until the game is turned off. So you need to manually reset them if needed.

- If you die and the level reloads, do you need to manually reset this back to the previous value?
- If you move to the next level, should this reset to zero or continue counting from where you left off?
- I think when you respawn from a checkpoint it's OK to keep the value, because those collectables will still be removed and this is a friendly to the player game design decision.

DavidJB526 replied:
Fixed in PlayerController Respawn function and CollectableManager.  CollectableManager sets public float startingCollectables equal to Collectable.collectables at Start.  This should only be at the start of each Scene.  PlayerController has a SerializeField for collectableUI, which then is called during Respawn function to set Collectable.collectables equal to startingCollectables.

This may have been a convoluted way to do this.

dantogno commented on Assets/Scripts/PlayerController.cs:
```diff
@@ -21,28 +41,31 @@ public class PlayerController : MonoBehaviour {
     [SerializeField]
-    private Text collectableText;
-    [SerializeField]
     private Slider overheatSlider;
```
Wrap all these serialized fields up in a #region https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/preprocessor-directives/preprocessor-region

dantogno reviewed (changes requested):
Great changes. Proud of how you addressed all the junk I came up with 👍.  I found more junk though 😜 

Also, with all these metas and prefabs changing, it makes me wonder if you used a diff version of Unity.

DavidJB526 committed:
Move SerializeFields in MouseTracker to top

DavidJB526 committed:
Clarified Variables in MouseTracker

DavidJB526 committed:
Clarified ToolTips in PlayerController

DavidJB526 committed:
Clarified required values in PlayerController ToolTips

DavidJB526 committed:
Add value clarifier for bulletSpeed in PlayerController

DavidJB526 committed:
Add Regions to PlayerController

DavidJB526 committed:
Removed one AudioSource and used SerializeFields for AudioClips instead

DavidJB526 committed:
Moved bool isActivated in Checkpoint Class

DavidJB526 committed:
Remove whitespace and add Summary to BulletController

DavidJB526 committed:
Set collectables to 0 at Start of CollectableManager

DavidJB526 committed:
Reset collectables to 0 if Player dies with no Checkpoint

DavidJB526 committed:
Remove setting collectables to 0 at Start of CollectableManager

DavidJB526 committed:
Created a starting value for CollectableManager and call that in PlayerController on Respawn

Should reset the static variable for collectables to

dantogno commented on Assets/Scripts/PlayerController.cs:
```diff
@@ -41,28 +44,32 @@ public class PlayerController : MonoBehaviour {
+    private AudioClip deathSound;
+    #endregion
 
+    #region Private Variables
```
The technical term for class-wide scope variables is "fields" (I don't know why). https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/classes-and-structs/fields

dantogno reviewed (approved):
Nice job, approved 👍
